GitFyle.Core.Api.Tests.Unit/Services/Foundations/Sources/SourceServiceTests.Exceptions.Modify.cs

```diff
@@ -0,0 +1,262 @@
+                broker.SelectSourceByIdAsync(randomSource.Id))
+                    .ThrowsAsync(dbUpdateException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(randomDateTimeOffset);
+
+            // when
+            ValueTask<Source> modifySourceTask =
+                this.sourceService.ModifySourceAsync(randomSource);
+
+            SourceDependencyException actualSourceDependencyException =
+                await Assert.ThrowsAsync<SourceDependencyException>(
+                    modifySourceTask.AsTask);
+
+            // then
+            actualSourceDependencyException.Should().BeEquivalentTo(
+                expectedSourceDependencyException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectSourceByIdAsync(randomSource.Id),
+                    Times.Once);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffsetAsync(),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogErrorAsync(It.Is(SameExceptionAs(
+                    expectedSourceDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        private async Task ShouldThrowDependencyValidationExceptionOnModifyIfDbUpdateConcurrencyOccursAndLogItAsync()
+        {
+            // given
+            int minutesInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+
+            Source randomSource =
+                CreateRandomModifySource(randomDateTimeOffset);
+
+            randomSource.CreatedDate = randomDateTimeOffset.AddMinutes(minutesInPast);
+
+            var dbUpdateConcurrencyException =
+                new DbUpdateConcurrencyException();
+
+            var lockedSourceException =
+                new LockedSourceException(
+                    message: "Locked source record error occurred, please try again.",
+                    innerException: dbUpdateConcurrencyException);
+
+            var expectedSourceDependencyValidationException =
+                new SourceDependencyValidationException(
+                    message: "Source dependency validation error occurred, fix errors and try again.",
+                    innerException: lockedSourceException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectSourceByIdAsync(randomSource.Id))
+                    .ThrowsAsync(dbUpdateConcurrencyException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffsetAsync())
+                    .ReturnsAsync(randomDateTimeOffset);
```

Comment:
We can remove the setup for dateTimeBrokerMock

```suggestion
            this.dateTimeBrokerMock.Setup(broker =>
                broker.GetCurrentDateTimeOffsetAsync())
                    .ReturnsAsync(dbUpdateConcurrencyException);
```